models/Asset/Image/Thumbnail/Processor.php

```diff
@@ -360,7 +371,7 @@ public static function process(Asset $asset, Config $config, $fileSystemPath = n
 
         $generated = true;
 
-        if ($contentOptimizedFormat) {
+        if ($contentOptimized) {
```

Comment:
Here would be the right place then for checking if an image should be optimized or not. 
But be careful here again! Pimcore supports custom optimizers based on the image type, so we can't hard-code this here. 
See also: `\Pimcore\Image\ImageOptimizerInterface`
Probably it's better to just add all images regardless of which type it is to the optimizer queue, if the 2 options mentioned before are not set.

Reply:
> Here would be the right place then for checking if an image should be optimized or not.

By now the optimization is always queued, unless some kind of preserving option (color, metadata) is set or it's clear the thumbnail is not intended for web (print, tiff).

However, I currently wonder about the forced webp format. I've added a comment there because I don't understand how it is okay for that to just swoop in and hijack the format as soon as it is supported by the adapter.
I see that there's some sort of handling in `Asset\Image\Thumbnail::getHtml()` . But again, `Controller\PublicServicesController::thumbnailAction()` doesn't handle that. So if understand correctly a direct http request to an unavailable thumbnail, and with `webp_auto_support` enabled - which it is by default, will overwrite the output format regardless of the requested image format.